### PR TITLE
pde_solver using neural_sde

### PR DESCRIPTION
**Pull request by KirillZubov**

### CODE REVIEW

File: src/neural_sde.jl [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+σᵀ∇u = Flux.Chain(Dense(d,hls,relu),
+                  Dense(hls,hls,relu),
+                  Dense(hls,d))
+g(X) = sum(X.^2)
+
+function trueODEfunc(dh,h,p,t)
+    u =  h[end]
+    X =  h[1:end-1]
+    _σᵀ∇u = σᵀ∇u(X) #TODO tracked σᵀ∇u
+    dh[end] = Flux.data(f(X, u, _σᵀ∇u, p, t))
```
Comment (ChrisRackauckas) [inline src/neural_sde.jl]: it's definitely going to fail if you `.data` this

File: src/neural_sde.jl [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    dh[1:end-1] .= μ(X,p,t)
+    dh
+end
+
+function true_noise_func(dh,h,p,t)
+    X = h[1:end-1]
+    _σᵀ∇u = Flux.data(σᵀ∇u(X))
+    dh[end] = sum(_σᵀ∇u) # TODO _σᵀ∇u*dW nondioganal
```
Comment (ChrisRackauckas) [inline src/neural_sde.jl]: yeah this isn't correct, but I think that works just for this case of sigma = 1?

File: src/neural_sde.jl [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    solve(prob, args...; kwargs...) |> Tracker.collect
+end
+
+n_sde = init_cond->neural_sde(init_cond,trueODEfunc,true_noise_func,tspan,SOSRI(),saveat=t,reltol=1e-1,abstol=1e-1)
+
+x0 = fill(1.0 , d)
+init_cond = fill(1.0 , d+1)
+
+function predict_n_sde()
+    _u0 = Flux.data(u0(x0)[1])
```
Comment (ChrisRackauckas) [inline src/neural_sde.jl]: The initial data has to be tracked. Take a look at the DiffEqFlux examples.

**Comment by ChrisRackauckas** [comment]
Yeah your issue is you're dropping all of the Tracker information, so there's no way for it to get a valid gradient.

File: src/neural_sde.jl (hunk 1 above)
Comment (KirillZubov) [reply src/neural_sde.jl]: yeah, I know. I fix this today

**Comment by KirillZubov** [comment]
> what does that stand for?

File: src/pde_solve_ns.jl [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+    data = Iterators.repeated((), maxiters)
+
+
+    #hidden layer
+    opt = alg.opt
+    u0 = alg.u0
+    σᵀ∇u = alg.σᵀ∇u
+    sde_algorithm = alg.sde_algorithm
+    ps = Flux.params(u0, σᵀ∇u...)
```
Comment (ChrisRackauckas) [inline src/pde_solve_ns.jl]:
```suggestion
    ps = Flux.params(u0, σᵀ∇u)
```

File: src/pde_solve_ns.jl [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    alg::NNPDENS;
+    timeseries_errors = true,
+    save_everystep=true,
+    adaptive=false,
+    abstol = 1f-6,
+    reltol = 1f-5,
+    verbose = false,
+    maxiters = 300,
+    dt,
+    trajectories)
```
Comment (ChrisRackauckas) [inline src/pde_solve_ns.jl]: splat kwargs to the sde solver

File: src/pde_solve_ns.jl [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    #hidden layer
+    opt = alg.opt
+    u0 = alg.u0
+    σᵀ∇u = alg.σᵀ∇u
+    sde_algorithm = alg.sde_algorithm
+    ps = Flux.params(u0, σᵀ∇u...)
+
+    function F(h, p, t)
+        u =  h[end]
+        X =  h[1:end-1].data
```
Comment (ChrisRackauckas) [inline src/pde_solve_ns.jl]: why are you taking the data part here?

File: Project.toml [hunk 7]
```diff
@@ -8,6 +8,9 @@ DiffEqBase = "2b5f629d-d688-5b77-993f-72d75c75574e"
 Flux = "587475ba-b771-5e3f-ad9e-33799f191a9c"
 Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
+StochasticDiffEq = "789caeaf-c7a9-5a7d-9973-96adeb23e2a0"
```
Comment (ChrisRackauckas) [inline Project.toml]: You don't need to directly depend on StochasticDiffEq

File: src/pde_solve_ns.jl (hunk 6 above)
Comment (KirillZubov) [reply src/pde_solve_ns.jl]: it is ```ERROR: LoadError: MethodError: no method matching Float32(::Tracker.TrackedReal{Float32})``` with case where σ or μ dependent from Tracked X

Comment (ChrisRackauckas) [reply src/pde_solve_ns.jl]: Interesting that it still works with this... but since it works it seems to be fine.

File: Project.toml (hunk 7 above)
Comment (KirillZubov) [reply Project.toml]: all the same, dependency on StochasticDiffEq  needed

Comment (KirillZubov) [reply Project.toml]: https://ci.appveyor.com/project/ChrisRackauckas/neuralnetdiffeq-jl/builds/26537921

**Pull request merged**